src/lib/Microsoft.Health.Fhir.Ingest/Service/NormalizationEventConsumerService.cs

```diff
@@ -0,0 +1,186 @@
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Health.Common.Telemetry;
+using Microsoft.Health.Events.EventConsumers;
+using Microsoft.Health.Events.Model;
+using Microsoft.Health.Events.Telemetry;
+using Microsoft.Health.Fhir.Ingest.Data;
+using Microsoft.Health.Fhir.Ingest.Telemetry;
+using Microsoft.Health.Fhir.Ingest.Template;
+using Microsoft.Health.Logging.Telemetry;
+using Newtonsoft.Json.Linq;
+using Polly;
+
+namespace Microsoft.Health.Fhir.Ingest.Service
+{
+    public class NormalizationEventConsumerService : IEventConsumer
+    {
+        private readonly IConverter<IEventMessage, JObject> _converter;
+        private readonly string _templateDefinition;
+        private readonly ITemplateManager _templateManager;
+        private readonly ITelemetryLogger _logger;
+        private readonly IEnumerableAsyncCollector<IMeasurement> _collector;
+        private readonly AsyncPolicy _retryPolicy;
+        private readonly CollectionTemplateFactory<IContentTemplate, IContentTemplate> _collectionTemplateFactory;
+        private readonly IExceptionTelemetryProcessor _exceptionTelemetryProcessor;
+
+        public NormalizationEventConsumerService(
+            IConverter<IEventMessage, JObject> converter,
+            string templateDefinition,
+            ITemplateManager templateManager,
+            IEnumerableAsyncCollector<IMeasurement> collector,
+            ITelemetryLogger logger,
+            CollectionTemplateFactory<IContentTemplate, IContentTemplate> collectionTemplateFactory,
+            NormalizationExceptionTelemetryProcessor exceptionTelemetryProcessor)
+        {
+            _converter = EnsureArg.IsNotNull(converter, nameof(converter));
+            _templateDefinition = EnsureArg.IsNotNullOrWhiteSpace(templateDefinition, nameof(templateDefinition));
+            _templateManager = EnsureArg.IsNotNull(templateManager, nameof(templateManager));
+            _collector = EnsureArg.IsNotNull(collector, nameof(collector));
+            _logger = EnsureArg.IsNotNull(logger, nameof(logger));
+            _retryPolicy = CreateRetryPolicy(logger);
+            _collectionTemplateFactory = EnsureArg.IsNotNull(collectionTemplateFactory, nameof(collectionTemplateFactory));
+            _exceptionTelemetryProcessor = EnsureArg.IsNotNull(exceptionTelemetryProcessor, nameof(exceptionTelemetryProcessor));
+
+            EventMetrics.SetConnectorOperation(ConnectorOperation.Normalization);
+        }
+
+        public async Task ConsumeAsync(IEnumerable<IEventMessage> events)
+        {
+            EnsureArg.IsNotNull(events);
+
+            await _retryPolicy.ExecuteAsync(async () => await ConsumeAsyncImpl(events));
+        }
+
+        private Task<IContentTemplate> GetNormalizationTemplate()

```

Comment:
One concern I have, is with a lower batch size (10), we are refreshing the template much more frequently.  Going to look at ways to reduce the impact.